bound/bound.go

```diff
@@ -36,3 +35,22 @@ func Intersection(bounds ...Bound) Bound {
 
 	return ret
 }
+
+// Union returns a contiguous union of the input bounds if possible.
+// Otherwise a NaN Bound is returned.
+func Union(bounds ...Bound) Bound {
+	if len(bounds) == 0 {
+		return Bound{Min: math.NaN(), Max: math.NaN()}
+	}
+
+	ret := Bound{Min: bounds[0].Min, Max: bounds[0].Max}
```

Comment:
We don't conventionally name variable based on whether they'll be returned. This might be, `u` or `union`.